language-server/src/main/scala/scala/meta/languageserver/Compiler.scala

```diff
@@ -71,6 +72,22 @@ class Compiler(
         .distinct
     }
   }
+
+  def typeAt(path: AbsolutePath, line: Int, column: Int): Option[String] = {
```

Comment:
How would you say the types compare to denotation.info? One thing we can do with denotation.info later down the road is to improve the pretty-printing so that some types are printed like `mutable.Set` and renamings in the source file are respected.

Reply:
See my other comment, it's pretty bad as of now.
The reason is that I'm taking the `.tpe` from the typed tree without analyzing which tree it is.
If I start discriminating on the trees it gets much better (e.g. if it's a `val` declaration, take the `tpe` of the rhs)

Reply:
I've just pushed a new commit, as an example of what I mean.

Reply:
Does this work for unsaved buffers that don't fully typecheck?